### PR TITLE
 detect: add requires keyword - v8

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: https://github.com/OISF/suricata/pull/10064

Changes from last PR:
- only accept version once as the new syntax doesn't require it multiple times
- add tenant-id to report message
- forward declaration to avoid void *
- actually allow unknown keywords as the comment suggests
- other minor cleanups that don't change the functionality

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1505

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17133

File: src/detect.h [hunk 1]
```diff
@@ -797,6 +800,7 @@ typedef struct SigFileLoaderStat_ {
     int total_files;
     int good_sigs_total;
     int bad_sigs_total;
+    int skipped_sigs_total;
```
Comment (victorjulien) [inline src/detect.h:803]: IIRC these stats can be polled over unix socket? Should we extend that support?

**Review by victorjulien — APPROVED** [review]

File: src/detect.h (hunk 1 above)
Comment (victorjulien) [reply src/detect.h:803]: can be done if follow up patch if needed

**Comment by catenacyber** [comment]
This looks awesome :-)

File: src/detect.h (hunk 1 above)
Comment (jasonish) [reply src/detect.h:803]: Appears to be in the JSON stat log (ie: rules_failed) that is then exportable over the socket.

Added https://redmine.openinfosecfoundation.org/issues/6637 and will be part of the backport.

**Comment by victorjulien** [comment]
Merged in #10076, thanks!

**Comment by catenacyber** [comment]
@jasonish oss-fuzz found a way to trigger the assert you added here 
```
 /* Now completely parse the rule. */
    ret = SigParse(de_ctx, sig, sigstr, dir, &parser, false);
    BUG_ON(ret == -4);
```

with this input
`drop ip :: 0 <> :: 0 requireS:feature;sid:1;`

Could you fix this ?

**Comment by jasonish** [comment]
> @jasonish oss-fuzz found a way to trigger the assert you added here
> 
> ```
>  /* Now completely parse the rule. */
>     ret = SigParse(de_ctx, sig, sigstr, dir, &parser, false);
>     BUG_ON(ret == -4);
> ```
> 
> with this input `drop ip :: 0 <> :: 0 requireS:feature;sid:1;`
> 
> Could you fix this ?

Is this different than https://github.com/OISF/suricata/pull/10112#pullrequestreview-1804279716?

**Comment by catenacyber** [comment]
> Is this different than [#10112 (review)](https://github.com/OISF/suricata/pull/10112#pullrequestreview-1804279716)?

Nope, thanks for the review

**Pull request closed without merging**